rakelib/artifacts.rake

```diff
@@ -127,14 +127,31 @@ namespace "artifact" do
     result
   end
 
-  # execute Kernel#system call,checking the exist status of the executed command and eventually reporting as exception
   def safe_system(*args)
-    if !system(*args)
-      status = $?
+    if args.first.is_a?(Hash) && args.length > 1
+      env_vars = args.shift
+      cmd = args.join(' ')
+      
+      # Format environment variables for command line
+      env_string = env_vars.map { |k, v| "#{k}=#{v}" }.join(' ')
+      full_cmd = "#{env_string} #{cmd}"
+      
+      output = `#{full_cmd} 2>&1`
```

Comment:
`Kernel#system` does not capture stdout/stderr so you lose any relevant failure messages which makes debugging (especially in CI) a nightmare. For some reason the open3 lib was not capturing these either. This was all I could come up with to get the info surfaced.

Reply:
yeah, this looks great. Updated. Thanks!